simulation/simulation.py

```diff
@@ -0,0 +1,210 @@
+        )
+        riskFreeTS = ql.YieldTermStructureHandle(
+            ql.FlatForward(self._params["start_date"],
+                           0.01, ql.Actual365Fixed())
+        )
+        volTS = ql.BlackVolTermStructureHandle(
+            ql.BlackConstantVol(
+                self._params["start_date"],
+                ql.NullCalendar(),
+                self._params["sigma"],
+                ql.Actual365Fixed(),
+            )
+        )
+
+        jumpIntensity = ql.QuoteHandle(ql.SimpleQuote(1.0))
+        jumpVolatility = ql.QuoteHandle(
+            ql.SimpleQuote(
+                self._params["sigma"] * np.sqrt(0.25 / jumpIntensity.value())
+            )
+        )
+        meanLogJump = ql.QuoteHandle(
+            ql.SimpleQuote(-jumpVolatility.value() * jumpVolatility.value())
+        )
+
+        return ql.Merton76Process(
+            self._params["initial_value"],
+            dividendTS,
+            riskFreeTS,
+            volTS,
+            jumpIntensity,
+            meanLogJump,
+            jumpVolatility,
+        )
+
+    def simulate(
+        self,
+        steps: int = 365,
+        maturity: int = 1,
+        n_simulations: int = 1000,
+        sigma: float = None,
+        mu: float = None,
+        initial_value: float = None,
+    ) -> None:
+        """
+        Given the time unit is days, the default arguments represent a path with a length of 1 year, consisting of 365 days.
+        1000 of those paths will be simulated.
+
+        Args:
+            steps (int, optional): Steps within a period of the maturity. Defaults to 365, which could be a year if the time unit is days.
+            maturity (int, optional): Maturity periods determining the length of the simulation. Defaults to 1.
+            n_simulations (int, optional): Number of paths to be simulated. Defaults to 1000.
+            sigma (float, optional): The standard deviation of the random processes. If None, it defaults to the standard deviation of the sample.
+            mu (float, optional): The mean drift of the random process. If None it defaults to the mean of the sample. 
+            initial_value (float, optional): The initial value of the random process. If None, it defaults to the initial value of the sample.
+
+        Returns:
+            None
+        """
+
+        self._params = {
+            "sigma": sigma if sigma else self.token_pair.returns.std()[0] * steps**0.5,
+            "mu": mu if mu else self.token_pair.returns.mean()[0]*365,
+            "initial_value": ql.QuoteHandle(
+                ql.SimpleQuote(
+                    initial_value if initial_value else self.token_pair.prices.iloc[0][0])
+            ),
+            "start_date": parse_date_to_quantlib(self.token_pair.prices.index[0]),
+            "total_steps": int(maturity * steps),
+            "_paths": [],
+        }
+
+        # TODO: Refactor this to allow more flexibility
+        if self.strategy == "GBM":
```

Comment:
`strategy` should be an enum, these conditionals should be replaces with a match syntax: https://blog.teclado.com/python-match-case/